worlds/stardew_valley/test/rules/TestCraftingRecipes.py

```diff
@@ -101,6 +118,23 @@ def test_can_craft_festival_recipe(self):
         self.collect([self.create_item("Progressive Season")] * 2)
         self.assert_rule_true(rule, self.multiworld.state)
 
+    def test_requires_mining_level_1_for_smelting_checks(self):
+        locations = ["Smelting", "Copper Pickaxe Upgrade", "Gold Trash Can Upgrade"]
+        rules = [self.world.logic.region.can_reach_location(location) for location in locations]
+        self.collect([self.create_item("Progressive Pickaxe")] * 4)
+        self.collect([self.create_item("Progressive Fishing Rod")] * 4)
+        self.collect([self.create_item("Progressive Sword")] * 4)
+        self.collect([self.create_item("Progressive Mine Elevator")] * 24)
+        self.collect([self.create_item("Progressive Trash Can")] * 2)
+        self.multiworld.state.collect(self.create_item("Furnace Recipe"), prevent_sweep=False)
+        self.collect([self.create_item("Combat Level")] * 10)
+        self.collect([self.create_item("Fishing Level")] * 10)
+        self.collect_all_the_money()
+        self.assert_rules_false(rules, self.multiworld.state)
+
+        self.collect([self.create_item("Mining Level")] * 10)
```

Comment:
Why is the test called `requires_mining_level_1` if it adds 10 mining levels? 🤔

Reply:
I can rename it. It requires mining level 1 for the furnace, but a higher level for the gold requirement of the asserted trash can